### PR TITLE
JBIDE-24954 migrate CDI, Deltaspike and SmartImport tests to RedDeer 2

### PR DESCRIPTION
**Pull request by luvalach**
JBIDE-24954 migrate CDI, Deltaspike and SmartImport tests to RedDeer 2. Remove org.jboss.tools.cdi.seam3.bot.test project.

### CODE REVIEW

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/cdi/ui/wizard/OpenCDINamedBeanDialog.java [hunk 1]
```diff
@@ -41,12 +41,12 @@ public void setMatchingItems(String... items) {
 	
 	public void ok() {
 		new OkButton().click();
-		new WaitWhile(new ShellWithTextIsActive("Open CDI Named Bean"));
+		new WaitWhile(new ShellIsActive("Open CDI Named Bean"));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/cdi/ui/wizard/OpenCDINamedBeanDialog.java]: use ShellIsAvailable

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/cdi/ui/wizard/OpenCDINamedBeanDialog.java [hunk 2]
```diff
@@ -41,12 +41,12 @@ public void setMatchingItems(String... items) {
...
 	}
 	
 	public void cancel() {
 		new CancelButton().click();
-		new WaitWhile(new ShellWithTextIsActive("Open CDI Named Bean"));
+		new WaitWhile(new ShellIsActive("Open CDI Named Bean"));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/cdi/ui/wizard/OpenCDINamedBeanDialog.java]: use ShellIsAvailable

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/cdi/ui/wizard/SpecifyBeanWizard.java [hunk 3]
```diff
@@ -109,7 +109,7 @@ private void selectInBeanQualifier(String qualifier) {
 	public void finish(){
 		String textShell = new DefaultShell().getText();
 		new PushButton("Finish").click();
-		new WaitWhile(new ShellWithTextIsActive(textShell));
+		new WaitWhile(new ShellIsActive(textShell));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/cdi/ui/wizard/SpecifyBeanWizard.java]: use ShellIsAvailable

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/AddIfClassAvailableDialog.java [hunk 4]
```diff
@@ -33,7 +33,7 @@ public void cancel() {
 	public void finish() {
 		String shellText = getText();
 		new PushButton("Finish").click();
-		new WaitWhile(new ShellWithTextIsActive(shellText));
+		new WaitWhile(new ShellIsActive(shellText));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/AddIfClassAvailableDialog.java]: use ShellIsAvailable(this)

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/AddIfSystemPropertyDialog.java [hunk 5]
```diff
@@ -39,7 +39,7 @@ public void cancel() {
 	public void finish() {
 		String shelltext = getText();
 		new PushButton("Finish").click();
-		new WaitWhile(new ShellWithTextIsActive(shelltext));
+		new WaitWhile(new ShellIsActive(shelltext));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/AddIfSystemPropertyDialog.java]: use ShellIsAvailable(this)

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/AddIncludeExcludeDialog.java [hunk 6]
```diff
@@ -50,7 +50,7 @@ public void cancel() {
 	public void finish() {
 		String shellText = getText();
 		new PushButton("Finish").click();
-		new WaitWhile(new ShellWithTextIsActive(shellText));
+		new WaitWhile(new ShellIsActive(shellText));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/AddIncludeExcludeDialog.java]: use ShellIsAvailable(this)

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/cdi/ui/NewQualifierCreationWizard.java [hunk 7]
```diff
@@ -39,7 +40,7 @@ public void finish() {
 		String shellText = new DefaultShell().getText();
 		Button button = new PushButton("Finish");
 		button.click();
-		new WaitWhile(new ShellWithTextIsActive(shellText), TimePeriod.LONG);
+		new WaitWhile(new ShellIsActive(shellText), TimePeriod.LONG);
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/cdi/ui/NewQualifierCreationWizard.java]: you can simply use super.finish(TImePeriod.LONG)

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/cdi/ui/wizard/CDIRefactorWizard.java [hunk 8]
```diff
@@ -28,7 +28,7 @@ public void next() {
 	public void finish() {
 		String shellText = new DefaultShell().getText();
 		new PushButton("Finish").click();
-		new WaitWhile(new ShellWithTextIsActive(shellText));
+		new WaitWhile(new ShellIsActive(shellText));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/cdi/ui/wizard/CDIRefactorWizard.java]: use ShellIsAvailable

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/editor/EditorPartWrapper.java [hunk 9]
```diff
@@ -193,7 +193,7 @@ private void editDefaultIncludeExclude(String oldName, String newName, boolean i
 	
 	private void removeIncludeExcludeWithHandler() {
 		new PushButton("Remove...").click();
-		new WaitUntil(new ShellWithTextIsAvailable("Confirmation"));
+		new WaitUntil(new ShellIsAvailable("Confirmation"));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/editor/EditorPartWrapper.java]: no need for the wait, DefaultShell already has the wait inside

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/editor/EditorPartWrapper.java [hunk 10]
```diff
@@ -227,8 +227,8 @@ public AddIfSystemPropertyDialog invokeWeldAddIfSystemPropertyDialog(String prop
 	}
 	
 	private AddIfSystemPropertyDialog invokeDefaultAddIfSystemPropertyDialog(String property){
-		new ContextMenu("Add System Property").select();
-		new WaitUntil(new ShellWithTextIsAvailable("Add If System Property"));
+		new ContextMenu().getItem("Add System Property").select();
+		new WaitUntil(new ShellIsAvailable("Add If System Property"));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/editor/EditorPartWrapper.java]: no need for the wait

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/editor/EditorPartWrapper.java [hunk 11]
```diff
@@ -244,8 +244,8 @@ public AddIncludeExcludeDialog invokeWeldAddIncludeExcludeDialog() {
 	}
 	
 	private AddIncludeExcludeDialog invokeDefaultAddIncludeExcludeDialog(){
-		new ContextMenu("Add Include/Exclude").select();
-		new WaitUntil(new ShellWithTextIsAvailable("Add Include/Exclude"));
+		new ContextMenu().getItem("Add Include/Exclude").select();
+		new WaitUntil(new ShellIsAvailable("Add Include/Exclude"));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/editor/EditorPartWrapper.java]: no need for the wait

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/editor/EditorPartWrapper.java [hunk 12]
```diff
@@ -262,8 +262,8 @@ public AddIfClassAvailableDialog invokeWeldAddClassAvailableDialog(String proper
 	}
 	
 	private AddIfClassAvailableDialog invokeDefaultWeldAddClassAvailableDialog(String property){
-		new ContextMenu("Add Class Available").select();
-		new WaitUntil(new ShellWithTextIsAvailable("Add If Class Available"));
+		new ContextMenu().getItem("Add Class Available").select();
+		new WaitUntil(new ShellIsAvailable("Add If Class Available"));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/common/model/ui/editor/EditorPartWrapper.java]: no need for the wait

File: plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/quickfix/QuickFixWizard.java [hunk 13]
```diff
@@ -65,7 +65,7 @@ public String getCDIQuickFix(String text) {
 	public void finish(){
 		String shellTitle = new DefaultShell().getText();
 		new PushButton("Finish").click();
-		new WaitWhile(new ShellWithTextIsActive(shellTitle));
+		new WaitWhile(new ShellIsActive(shellTitle));
```
Comment (rawagner) [inline plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/quickfix/QuickFixWizard.java]: use ShellIsAvailable

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDITestBase.java [hunk 14]
```diff
@@ -73,93 +75,118 @@ public void prepareWorkspace() {
...
 		preferenceDialog.open();
-		SourceLookupPreferencePage sourceLookupPreferencePage = new SourceLookupPreferencePage();
+		SourceLookupPreferencePage sourceLookupPreferencePage = new SourceLookupPreferencePage(preferenceDialog);
 		preferenceDialog.select(sourceLookupPreferencePage);
-		sourceLookupPreferencePage.setSourceAttachment(
-				SourceLookupPreferencePage.SourceAttachmentEnum.NEVER);
+		sourceLookupPreferencePage.setSourceAttachment(SourceLookupPreferencePage.SourceAttachmentEnum.NEVER);
 		preferenceDialog.ok();
-		new WaitUntil(new ShellWithTextIsActive(originalShellText));
+		new WaitUntil(new ShellIsActive(originalShellText));
```
Comment (rawagner) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDITestBase.java]: ShellIsAvailable

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/validation/template/CDIValidatorTemplate.java [hunk 15]
```diff
@@ -94,17 +98,17 @@ private static void modifyCDIValidatorState(boolean enable) {
...
 	}
 
 	private static void closeSettingsChangedShell() {
-		new WaitUntil(new ShellWithTextIsAvailable("Validator Settings Changed"), TimePeriod.LONG);
+		new WaitUntil(new ShellIsAvailable("Validator Settings Changed"), TimePeriod.LONG);
 		DefaultShell shell = new DefaultShell("Validator Settings Changed");
 		String shellText = shell.getText();
 		new PushButton("Yes").click();
-		new WaitWhile(new ShellWithTextIsActive(shellText), TimePeriod.LONG);
+		new WaitWhile(new ShellIsActive(shellText), TimePeriod.LONG);
```
Comment (rawagner) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/validation/template/CDIValidatorTemplate.java]: ShellIsAvailable

File: tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java [hunk 16]
```diff
@@ -217,10 +219,10 @@ protected void openClass(String projectName, String packageName, String classFul
 	}
 
 	protected static void cleanProjects() {
-		new ShellMenu(new WithTextMatcher("Project"), new RegexMatcher("Clean.*")).select();
-		new WaitUntil(new ShellWithTextIsActive("Clean"));
+		new ShellMenuItem(new WithTextMatcher("Project"), new RegexMatcher("Clean.*")).select();
+		new WaitUntil(new ShellIsActive("Clean"));
```
Comment (rawagner) [inline tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java]: ShellIsAvailable

File: tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java [hunk 17]
```diff
@@ -217,10 +219,10 @@ protected void openClass(String projectName, String packageName, String classFul
...
 		new PushButton("OK").click();
-		new WaitWhile(new ShellWithTextIsActive("Clean"));
+		new WaitWhile(new ShellIsActive("Clean"));
```
Comment (rawagner) [inline tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java]: ShellIsAvailable

File: site/category.xml [hunk 18]
```diff
@@ -75,7 +75,6 @@
 <bundle id="org.jboss.tools.seam.reddeer.source"><category name="JBoss Tools Core - Integration Test Sources"/></bundle>
 <bundle id="org.jboss.tools.aerogear.ui.bot.test.source"><category name="JBoss Tools Core - Integration Test Sources"/></bundle>
 <bundle id="org.jboss.tools.cdi.bot.test.source"><category name="JBoss Tools Core - Integration Test Sources"/></bundle>
-<bundle id="org.jboss.tools.cdi.seam3.bot.test.source"><category name="JBoss Tools Core - Integration Test Sources"/></bundle>
```
Comment (rawagner) [inline site/category.xml]: remove  org.jboss.tools.cdi.seam3.bot.test too

**Comment by luvalach** [comment]
Thanks for review @rawagner, could you check it again? Everything you were complaining to should be fixed.

**Pull request merged**